rest-service/manager_rest/test/endpoints/test_plugins_updates.py

```diff
@@ -0,0 +1,352 @@
+#########
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+import uuid
+from copy import deepcopy
+
+import unittest
+from mock import patch
+
+from cloudify_rest_client.exceptions import CloudifyClientError
+
+from cloudify.models_states import ExecutionState
+
+from dsl_parser.constants import (PLUGIN_NAME_KEY,
+                                  WORKFLOW_PLUGINS_TO_INSTALL,
+                                  DEPLOYMENT_PLUGINS_TO_INSTALL)
+
+from manager_rest.test import base_test
+from manager_rest.test.attribute import attr
+from manager_rest.test.base_test import BaseServerTestCase
+
+from manager_rest.plugin_update.constants import STATES
+from manager_rest.manager_exceptions import NotFoundError
+from manager_rest.storage import get_storage_manager, models
+from manager_rest.plugin_update.manager import \
+    (PluginsUpdateManager,
+     _did_plugins_to_install_change as plugins_change_detector)
+
+
+def _plugins_update_test_setup(self):
```

Comment:
Create a base test class and inherit from that, by this u will save a couple of lines and the code will look better